packages/@react-spectrum/color/src/ColorSwatchPicker.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {ColorSwatchPicker as AriaColorSwatchPicker, ColorSwatchPickerItem as AriaColorSwatchPickerItem} from 'react-aria-components';
+import {Color} from '@react-types/color';
+import {DOMRef, StyleProps, ValueBase} from '@react-types/shared';
+import React, {forwardRef, ReactElement, ReactNode} from 'react';
+import {SpectrumColorSwatchContext, SpectrumColorSwatchProps} from './ColorSwatch';
+import {style} from '@react-spectrum/style-macro-s1' with {type: 'macro'};
+import {useDOMRef, useStyleProps} from '@react-spectrum/utils';
+
+export interface SpectrumColorSwatchPickerProps extends ValueBase<string | Color | null, Color>, StyleProps {
+  /** The ColorSwatches within the ColorSwatchPicker. */
+  children: ReactNode,
+  /**
+   * The amount of padding between the swatches.
+   * @default "regular"
+   */
+  density?: 'compact' | 'regular' | 'spacious',
+  /**
+   * The size of the color swatches.
+   * @default "M"
+   */
+  size?: 'XS' | 'S' | 'M' | 'L',
+  /**
+   * The corner rounding of the color swatches.
+   * @default "none"
+   */
+  rounding?: 'none' | 'default' | 'full'
```

Comment:
I followed spectrum's options here, but it is sorta weird that density uses "regular" and rounding uses "default"...